src/fsharp/IlxGen.fs

```diff
@@ -817,7 +817,7 @@ let GetMethodSpecForMemberVal amap g (memberInfo:ValMemberInfo) (vref:ValRef) =
 
         let thisTy = if isByrefTy g thisTy then destByrefTy g thisTy else thisTy
         let thisArgTys = argsOfAppTy g thisTy
-        if ctps.Length <> thisArgTys.Length then
+        if numParentTypars <> thisArgTys.Length then
            warning(InternalError(sprintf "CodeGen check: type checking did not quantify the correct number of type variables for this method, #parentTypars = %d, #ctps = %d, #mtps = %d, #thisArgTys = %d" numParentTypars ctps.Length mtps.Length thisArgTys.Length,m))
```

Comment:
If this change is correct, you can also replace ctps.Length in the message itself.

> let numParentTypars = parentTypars.Length

numParentTypars is not identical to ctps.Length though, at least not by quickly looking at the code so I'm not convinced you can replace it in the if.

Reply:
done

Reply:
see `let ctps,mtps = List.chop numParentTypars tps`

Reply:
thanks, ok so ctps length will not be more than numParentTypars but can it be less? apparently no because of 

> if tps.Length < numParentTypars then error

so it looks valid :slightly_smiling_face: